osmnx/projection.py

```diff
@@ -105,8 +105,8 @@ def project_gdf(gdf, to_crs=None, to_latlong=False):
         if is_projected(gdf.crs):  # pragma: no cover
             raise ValueError("Geometry must be unprojected to calculate UTM zone")
 
-        # calculate longitude of centroid of union of all geometries in gdf
-        avg_lng = gdf["geometry"].unary_union.centroid.x
+        # calculate approximate longitude of centroid of union of all geometries in gdf
+        avg_lng = gdf["geometry"].to_crs("epsg:4326").representative_point().x.mean()
```

Comment:
The `.to_crs("epsg:4326")` should be unnecessary as the lines above verify that the geometries are unprojected. The function should be able to handle geometries in any unprojected CRS without first requiring reprojection to epsg:4326.

Reply:
I see. I thought it needs to be in 4326.